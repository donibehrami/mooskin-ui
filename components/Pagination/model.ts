import { IDivBoxComponentProps } from '../Box/model';

export interface IPaginationComponentProps extends IDivBoxComponentProps {
    /** number of the active button page */
    activePage: number;

    /** onclick callback when the button is clicked */
    onClickButton?: (e: React.MouseEvent<HTMLElement>, value: number) => void;

    children?: React.ReactElement<IPaginationButtonComponentProps> | Array<React.ReactElement<IPaginationButtonComponentProps>>;
}

export interface IPaginationButtonComponentProps extends IDivBoxComponentProps {
    /** whether the current button is active */
    active?: boolean;
}
