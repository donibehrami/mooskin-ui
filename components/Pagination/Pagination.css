@import '../_utils/globals/variables.css';
@import '../_utils/globals/screens.css';


.pagination{
    display: flex;
    flex-direction: row;
    justify-content: center;
}

.paginationInner{
    display: flex;
    flex-direction: row;
    justify-content: center;
    border-radius: 3px;
}

.paginationButton{
    width: 40px;
    height:40px;
    justify-content: center;
    display: flex;
    align-items: center;
    cursor: pointer;
    color: var(--inputFontColor);
    border-radius: 3px; 
    user-select: none;
}

.paginationButton:hover{
    background: #eee;
}

.activeButton {
    background: var(--moosendBlue);
    color: #fff;
}

.activeButton:hover{
    background: var(--moosendBlue);
}

.disabledButton {
  pointer-events: none;
}

@media (--xSmall){

    .itemButton{
        display: none;
    }

    .disabledButton{
        display: none;
    }

    .activeButton{
        display: inherit;
    }

}