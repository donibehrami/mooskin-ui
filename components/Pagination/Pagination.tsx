import * as React from 'react';

import styles from './Pagination.css';

export interface IPaginationProps {

    /** custom classes for Pagination */
    className?: string;

    /** custom id for Pagination */
    id?: string;

    /** custom styles for Pagination top element */
    style?: React.CSSProperties;

    /** total number of items */
    items: number;

    /** the number of the current page */
    currentItem?: number;

    /** the max number of buttons to be displayed */
    maxButtons?: number;

    /** boolean to specify whether to show the next item button */
    nextBtn?: boolean;

    /** boolean to specify whether to show the previous item button */
    prevBtn?: boolean;

    /** boolean to specify whether to show the very last item button */
    lastBtn?: boolean;

    /** boolean to specify whether to show the very first item button */
    firstBtn?: boolean;

    /** callback that is called when a button is clicked */
    onClick: (item: number) => void;
}

export interface IPaginationButtonProps {
    active: boolean;
    disabled?: boolean;
    label: string; // could be an icon
    item: number;
    className?: string;
    onClick: (item: number) => (e: React.MouseEvent<HTMLElement>) => void;
}

export enum RendereableItemTypes {
    right,
    left,
    current,
    none
}

export default class Pagination extends React.Component<IPaginationProps, {}>{

    static defaultProps: Partial<IPaginationProps> = {
        currentItem: 1,
        firstBtn: false,
        lastBtn: false,
        maxButtons: 5,
        nextBtn: false,
        prevBtn: false,
    };

    render() {

        const {className, style, id} = this.props;

        return (
            <div
                className={`${styles.pagination} ${className}`}
                style={style}
                id={id}
            >
                <div className={styles.paginationInner}>
                    {this.renderAllButtons()}
                </div>
            </div>
        );
    }

    onClick = (item: number) => {
        return (e: React.MouseEvent<HTMLElement>) => {
            this.props.onClick(item);
        };
    }

    renderAllButtons(){
        const {currentItem, items, firstBtn, lastBtn, nextBtn, prevBtn} = this.props;

        // render back and prev buttons as well as ellipsis as needed
        return currentItem && [
            firstBtn &&
            (
                <PaginationButton
                    key={'asd1'}
                    active={false}
                    label={'<<'}
                    item={1}
                    onClick={this.onClick}
                    className="pagination-first-btn"
                />
            ),
            prevBtn &&
            (
                <PaginationButton
                    key={'asd2'}
                    active={false}
                    label={'<'}
                    item={currentItem && currentItem - 1 || 1}
                    onClick={this.onClick}
                    className="pagination-prev-btn"
                />
            ),
            this.needsLeftEllipsis() &&
            (
                <PaginationButton
                    key={'asdd2'}
                    active={false}
                    label={'...'}
                    item={currentItem}
                    onClick={(this.onClick)}
                    disabled
                    className="pagination-ellipsis"
                />
            ),
            this.renderNormalButtons(),
            this.needsRightEllipsis() &&
            (
                <PaginationButton
                    key={'asdd3'}
                    active={false}
                    label={'...'}
                    item={currentItem}
                    onClick={(this.onClick)}
                    disabled
                    className="pagination-ellipsis"
                />
            ),
            nextBtn &&
            (
                <PaginationButton
                    key={'asd3'}
                    active={false}
                    label={'>'}
                    item={currentItem && currentItem + 1 || items}
                    onClick={this.onClick}
                    className="pagination-next-btn"
                />
            ),
            lastBtn &&
            (
                <PaginationButton
                    key={'asd4'}
                    active={false}
                    label={'>>'}
                    item={items}
                    onClick={this.onClick}
                    className="pagination-last-btn"
                />
            ),
        ];
    }

    renderNormalButtons(){

        const buttonCounts: [number, number] = this.getLeftRightButtonCounts();

        return [...Array(this.props.items).keys()].map((index) => {

            const item = index + 1;

            const itemType = this.getItemType(item, buttonCounts[0], buttonCounts[1]);

            return itemType !== RendereableItemTypes.none && (
                <PaginationButton
                    key={item}
                    active={itemType === RendereableItemTypes.current}
                    label={item.toString()}
                    item={item}
                    onClick={this.onClick}
                    className="pagination-normal-btn"
                />
            );
        });
    }

    getItemType(item: number, leftButtonsCount: number, rightButtonsCount: number){

        const {currentItem} = this.props;

        let itemType: RendereableItemTypes = RendereableItemTypes.none;

        // using some minor math with button counts derived from maxButtons check if item should be rendered
        if (currentItem && item < currentItem && item + leftButtonsCount >= currentItem){
            itemType = RendereableItemTypes.left;
        }else if (currentItem && item > currentItem && item - rightButtonsCount <= currentItem){
            itemType = RendereableItemTypes.right;
        }else if (currentItem && item === currentItem){
            itemType = RendereableItemTypes.current;
        }

        return itemType;
    }

    getLeftRightButtonCounts(): [number, number]{

        const {maxButtons} = this.props;

        let leftButtonsCount: number = 0;
        let rightButtonsCount: number = 0;

        // determine button counts to the left and right of the active item, based on if the number is even or odd
        if (maxButtons){
            leftButtonsCount = maxButtons % 2 !== 0 ? (maxButtons - 1) / 2 : maxButtons / 2;
            rightButtonsCount = maxButtons % 2 !== 0 ? (maxButtons - 1) / 2 : leftButtonsCount - 1;
        }

        return [leftButtonsCount, rightButtonsCount];
    }

    needsLeftEllipsis(){
        const {currentItem} = this.props;
        const leftButtonCount = this.getLeftRightButtonCounts()[0];
        return currentItem && currentItem - leftButtonCount > 1;
    }

    needsRightEllipsis(){
        const {currentItem, items} = this.props;
        const rightButtonCount = this.getLeftRightButtonCounts()[1];
        return currentItem && currentItem + rightButtonCount < items;
    }
}

export const PaginationButton: React.StatelessComponent<IPaginationButtonProps> = (props) => {

    const activeClass = props.active ? styles.activeButton : '';
    const disabledClass = props.disabled ? styles.disabledButton : '';

    return (
        <div
            className={`${styles.paginationButton} ${activeClass} ${disabledClass} ${props.className || ''}`}
            onClick={props.onClick(props.item)}
        >
            {props.label}
        </div>
    );
};
