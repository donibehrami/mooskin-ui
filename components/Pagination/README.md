# MooSkin UI - Pagination Component

___

### Usage

To start using the Pagination Component first you have to Import it

```
Import {Pagination} from 'mooskin';
```
or modular import
```
// JS
import Pagination from 'mooskin/lib/Pagination';

// CSS
import 'mooskin/lib/Pagination/style.css';
```

And then you can simply start using it by typing

```
<Pagination attribute1="atr" attribute2="asd" />
```

### Examples


Adding prev and next buttons

```
<Pagination items={5} prevBtn nextBtn />
```

or pass it a callback for event handling when a user clicks on a button

```
<Pagination items={20} onClick={yourFunc} />
```

or just give it a custom style

```
<Pagination items={20} style={yourStyle} />
```

<div class="playground-doc">

## Supported attributes

* `items` - (required) number of pages to be paginated
* `currentItem` - (required) indicates the current page, defaults to 1
* `maxButtons` - indicates the max number of buttons to be dispayed, defaults to 5 
* `nextBtn` - boolean, whether to show the 'next' button, defaults to false
* `prevBtn` - boolean, whether to show the 'prev' button, defaults to false
* `firstBtn` - boolean, whether to show the 'navigate to first' button, defaults to false
* `lastBtn` - boolean, whether to show the 'navigate to last' button, defaults to false
* `onClick` - callback to be triggered on button click, takes item number as an argument
* `id` - main element id
* `inverseStyle` - inverse stying for pagination
* `className` - main element css class
* `style` - main element style

</div>

Allthough these attributes are supported, most of them are optional.


#### For more

___

[MooSkin-UI](https://github.com/moosend/mooskin-ui)