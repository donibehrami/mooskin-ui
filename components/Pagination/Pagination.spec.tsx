import * as React from 'react';
import Pagination from './Pagination';

import {render, shallow} from 'enzyme';

describe('Pagination', () => {

    test('renders correctly', () => {
        const func = jest.fn();

        const tree = shallow(
            <Pagination items={10} onClick={() => 'asd'}/>
        );

        expect(tree).toMatchSnapshot();
    });
});
