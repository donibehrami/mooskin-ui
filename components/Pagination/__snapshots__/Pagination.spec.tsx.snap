// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders correctly 1`] = `
<Pagination
  activePage={3}
  className=""
  onClickButton={[MockFunction]}
  style={Object {}}
>
  <StyledPagination
    activePage={3}
    className=""
    onClickButton={[MockFunction]}
    style={Object {}}
  >
    <Box
      activePage={3}
      className="sc-gtssRu bnocXT"
      onClickButton={[MockFunction]}
      style={Object {}}
    >
      <Component
        activePage={3}
        className="sc-gtssRu bnocXT"
        onClickButton={[MockFunction]}
        palette={
          Object {
            "actionsDropdown": Object {
              "backgroundColor": "#3fbaca",
              "fontColor": "#ffffff",
            },
            "anchor": Object {
              "disabledFontColor": "#9d9d9d",
              "fontColor": "#3fbaca",
            },
            "button": Object {
              "backgroundColor": "#3fbaca",
              "disabledBackgroundColor": "#e2e2e2",
              "fontColor": "#ffffff",
            },
            "commonElement": Object {
              "backgroundColor": "#ffffff",
              "borderColor": "#e2e2e2",
              "fontColor": "#293346",
            },
            "description": Object {
              "fontColor": "#9d9d9d",
            },
            "label": Object {
              "fontColor": "#2d2d2d",
            },
            "layout": Object {
              "backgroundColor": "#ffffff",
              "borderColor": "#e2e2e2",
              "fontColor": "#293346",
            },
            "main": Object {
              "backgroundColor": "#f7f7f7",
            },
            "toggleElement": Object {
              "backgroundColor": "#425579",
              "borderColor": "#e2e2e2",
              "fontColor": "#ffffff",
              "secondaryBackgroundColor": "#ffffff",
              "secondaryFontColor": "#9d9d9d",
            },
          }
        }
        style={Object {}}
      >
        <StyledBox
          activePage={3}
          className="sc-gtssRu bnocXT"
          onClickButton={[MockFunction]}
          palette={
            Object {
              "actionsDropdown": Object {
                "backgroundColor": "#3fbaca",
                "fontColor": "#ffffff",
              },
              "anchor": Object {
                "disabledFontColor": "#9d9d9d",
                "fontColor": "#3fbaca",
              },
              "button": Object {
                "backgroundColor": "#3fbaca",
                "disabledBackgroundColor": "#e2e2e2",
                "fontColor": "#ffffff",
              },
              "commonElement": Object {
                "backgroundColor": "#ffffff",
                "borderColor": "#e2e2e2",
                "fontColor": "#293346",
              },
              "description": Object {
                "fontColor": "#9d9d9d",
              },
              "label": Object {
                "fontColor": "#2d2d2d",
              },
              "layout": Object {
                "backgroundColor": "#ffffff",
                "borderColor": "#e2e2e2",
                "fontColor": "#293346",
              },
              "main": Object {
                "backgroundColor": "#f7f7f7",
              },
              "toggleElement": Object {
                "backgroundColor": "#425579",
                "borderColor": "#e2e2e2",
                "fontColor": "#ffffff",
                "secondaryBackgroundColor": "#ffffff",
                "secondaryFontColor": "#9d9d9d",
              },
            }
          }
          style={Object {}}
        >
          <div
            className="sc-bdnylx dOPKTt sc-gtssRu bnocXT"
            onClickButton={[MockFunction]}
            style={Object {}}
          >
            <PaginationButton
              active={false}
              className=""
              key="0/.0"
              onClick={[Function]}
              style={Object {}}
            >
              <StyledPaginationButton
                active={false}
                className=""
                onClick={[Function]}
                style={Object {}}
              >
                <Box
                  active={false}
                  className="sc-dlnjPT gKiNcx"
                  onClick={[Function]}
                  style={Object {}}
                >
                  <Component
                    active={false}
                    className="sc-dlnjPT gKiNcx"
                    onClick={[Function]}
                    palette={
                      Object {
                        "actionsDropdown": Object {
                          "backgroundColor": "#3fbaca",
                          "fontColor": "#ffffff",
                        },
                        "anchor": Object {
                          "disabledFontColor": "#9d9d9d",
                          "fontColor": "#3fbaca",
                        },
                        "button": Object {
                          "backgroundColor": "#3fbaca",
                          "disabledBackgroundColor": "#e2e2e2",
                          "fontColor": "#ffffff",
                        },
                        "commonElement": Object {
                          "backgroundColor": "#ffffff",
                          "borderColor": "#e2e2e2",
                          "fontColor": "#293346",
                        },
                        "description": Object {
                          "fontColor": "#9d9d9d",
                        },
                        "label": Object {
                          "fontColor": "#2d2d2d",
                        },
                        "layout": Object {
                          "backgroundColor": "#ffffff",
                          "borderColor": "#e2e2e2",
                          "fontColor": "#293346",
                        },
                        "main": Object {
                          "backgroundColor": "#f7f7f7",
                        },
                        "toggleElement": Object {
                          "backgroundColor": "#425579",
                          "borderColor": "#e2e2e2",
                          "fontColor": "#ffffff",
                          "secondaryBackgroundColor": "#ffffff",
                          "secondaryFontColor": "#9d9d9d",
                        },
                      }
                    }
                    style={Object {}}
                  >
                    <StyledBox
                      active={false}
                      className="sc-dlnjPT gKiNcx"
                      onClick={[Function]}
                      palette={
                        Object {
                          "actionsDropdown": Object {
                            "backgroundColor": "#3fbaca",
                            "fontColor": "#ffffff",
                          },
                          "anchor": Object {
                            "disabledFontColor": "#9d9d9d",
                            "fontColor": "#3fbaca",
                          },
                          "button": Object {
                            "backgroundColor": "#3fbaca",
                            "disabledBackgroundColor": "#e2e2e2",
                            "fontColor": "#ffffff",
                          },
                          "commonElement": Object {
                            "backgroundColor": "#ffffff",
                            "borderColor": "#e2e2e2",
                            "fontColor": "#293346",
                          },
                          "description": Object {
                            "fontColor": "#9d9d9d",
                          },
                          "label": Object {
                            "fontColor": "#2d2d2d",
                          },
                          "layout": Object {
                            "backgroundColor": "#ffffff",
                            "borderColor": "#e2e2e2",
                            "fontColor": "#293346",
                          },
                          "main": Object {
                            "backgroundColor": "#f7f7f7",
                          },
                          "toggleElement": Object {
                            "backgroundColor": "#425579",
                            "borderColor": "#e2e2e2",
                            "fontColor": "#ffffff",
                            "secondaryBackgroundColor": "#ffffff",
                            "secondaryFontColor": "#9d9d9d",
                          },
                        }
                      }
                      style={Object {}}
                    >
                      <div
                        className="sc-bdnylx dOPKTt sc-dlnjPT gKiNcx"
                        onClick={[Function]}
                        style={Object {}}
                      >
                        1
                      </div>
                    </StyledBox>
                  </Component>
                </Box>
              </StyledPaginationButton>
            </PaginationButton>
            <PaginationButton
              active={false}
              className=""
              key="1/.1"
              onClick={[Function]}
              style={Object {}}
            >
              <StyledPaginationButton
                active={false}
                className=""
                onClick={[Function]}
                style={Object {}}
              >
                <Box
                  active={false}
                  className="sc-dlnjPT gKiNcx"
                  onClick={[Function]}
                  style={Object {}}
                >
                  <Component
                    active={false}
                    className="sc-dlnjPT gKiNcx"
                    onClick={[Function]}
                    palette={
                      Object {
                        "actionsDropdown": Object {
                          "backgroundColor": "#3fbaca",
                          "fontColor": "#ffffff",
                        },
                        "anchor": Object {
                          "disabledFontColor": "#9d9d9d",
                          "fontColor": "#3fbaca",
                        },
                        "button": Object {
                          "backgroundColor": "#3fbaca",
                          "disabledBackgroundColor": "#e2e2e2",
                          "fontColor": "#ffffff",
                        },
                        "commonElement": Object {
                          "backgroundColor": "#ffffff",
                          "borderColor": "#e2e2e2",
                          "fontColor": "#293346",
                        },
                        "description": Object {
                          "fontColor": "#9d9d9d",
                        },
                        "label": Object {
                          "fontColor": "#2d2d2d",
                        },
                        "layout": Object {
                          "backgroundColor": "#ffffff",
                          "borderColor": "#e2e2e2",
                          "fontColor": "#293346",
                        },
                        "main": Object {
                          "backgroundColor": "#f7f7f7",
                        },
                        "toggleElement": Object {
                          "backgroundColor": "#425579",
                          "borderColor": "#e2e2e2",
                          "fontColor": "#ffffff",
                          "secondaryBackgroundColor": "#ffffff",
                          "secondaryFontColor": "#9d9d9d",
                        },
                      }
                    }
                    style={Object {}}
                  >
                    <StyledBox
                      active={false}
                      className="sc-dlnjPT gKiNcx"
                      onClick={[Function]}
                      palette={
                        Object {
                          "actionsDropdown": Object {
                            "backgroundColor": "#3fbaca",
                            "fontColor": "#ffffff",
                          },
                          "anchor": Object {
                            "disabledFontColor": "#9d9d9d",
                            "fontColor": "#3fbaca",
                          },
                          "button": Object {
                            "backgroundColor": "#3fbaca",
                            "disabledBackgroundColor": "#e2e2e2",
                            "fontColor": "#ffffff",
                          },
                          "commonElement": Object {
                            "backgroundColor": "#ffffff",
                            "borderColor": "#e2e2e2",
                            "fontColor": "#293346",
                          },
                          "description": Object {
                            "fontColor": "#9d9d9d",
                          },
                          "label": Object {
                            "fontColor": "#2d2d2d",
                          },
                          "layout": Object {
                            "backgroundColor": "#ffffff",
                            "borderColor": "#e2e2e2",
                            "fontColor": "#293346",
                          },
                          "main": Object {
                            "backgroundColor": "#f7f7f7",
                          },
                          "toggleElement": Object {
                            "backgroundColor": "#425579",
                            "borderColor": "#e2e2e2",
                            "fontColor": "#ffffff",
                            "secondaryBackgroundColor": "#ffffff",
                            "secondaryFontColor": "#9d9d9d",
                          },
                        }
                      }
                      style={Object {}}
                    >
                      <div
                        className="sc-bdnylx dOPKTt sc-dlnjPT gKiNcx"
                        onClick={[Function]}
                        style={Object {}}
                      >
                        2
                      </div>
                    </StyledBox>
                  </Component>
                </Box>
              </StyledPaginationButton>
            </PaginationButton>
            <PaginationButton
              active={true}
              className=""
              key="2/.2"
              onClick={[Function]}
              style={Object {}}
            >
              <StyledPaginationButton
                active={true}
                className=""
                onClick={[Function]}
                style={Object {}}
              >
                <Box
                  active={true}
                  className="sc-dlnjPT cUoMmY"
                  onClick={[Function]}
                  style={Object {}}
                >
                  <Component
                    active={true}
                    className="sc-dlnjPT cUoMmY"
                    onClick={[Function]}
                    palette={
                      Object {
                        "actionsDropdown": Object {
                          "backgroundColor": "#3fbaca",
                          "fontColor": "#ffffff",
                        },
                        "anchor": Object {
                          "disabledFontColor": "#9d9d9d",
                          "fontColor": "#3fbaca",
                        },
                        "button": Object {
                          "backgroundColor": "#3fbaca",
                          "disabledBackgroundColor": "#e2e2e2",
                          "fontColor": "#ffffff",
                        },
                        "commonElement": Object {
                          "backgroundColor": "#ffffff",
                          "borderColor": "#e2e2e2",
                          "fontColor": "#293346",
                        },
                        "description": Object {
                          "fontColor": "#9d9d9d",
                        },
                        "label": Object {
                          "fontColor": "#2d2d2d",
                        },
                        "layout": Object {
                          "backgroundColor": "#ffffff",
                          "borderColor": "#e2e2e2",
                          "fontColor": "#293346",
                        },
                        "main": Object {
                          "backgroundColor": "#f7f7f7",
                        },
                        "toggleElement": Object {
                          "backgroundColor": "#425579",
                          "borderColor": "#e2e2e2",
                          "fontColor": "#ffffff",
                          "secondaryBackgroundColor": "#ffffff",
                          "secondaryFontColor": "#9d9d9d",
                        },
                      }
                    }
                    style={Object {}}
                  >
                    <StyledBox
                      active={true}
                      className="sc-dlnjPT cUoMmY"
                      onClick={[Function]}
                      palette={
                        Object {
                          "actionsDropdown": Object {
                            "backgroundColor": "#3fbaca",
                            "fontColor": "#ffffff",
                          },
                          "anchor": Object {
                            "disabledFontColor": "#9d9d9d",
                            "fontColor": "#3fbaca",
                          },
                          "button": Object {
                            "backgroundColor": "#3fbaca",
                            "disabledBackgroundColor": "#e2e2e2",
                            "fontColor": "#ffffff",
                          },
                          "commonElement": Object {
                            "backgroundColor": "#ffffff",
                            "borderColor": "#e2e2e2",
                            "fontColor": "#293346",
                          },
                          "description": Object {
                            "fontColor": "#9d9d9d",
                          },
                          "label": Object {
                            "fontColor": "#2d2d2d",
                          },
                          "layout": Object {
                            "backgroundColor": "#ffffff",
                            "borderColor": "#e2e2e2",
                            "fontColor": "#293346",
                          },
                          "main": Object {
                            "backgroundColor": "#f7f7f7",
                          },
                          "toggleElement": Object {
                            "backgroundColor": "#425579",
                            "borderColor": "#e2e2e2",
                            "fontColor": "#ffffff",
                            "secondaryBackgroundColor": "#ffffff",
                            "secondaryFontColor": "#9d9d9d",
                          },
                        }
                      }
                      style={Object {}}
                    >
                      <div
                        className="sc-bdnylx dOPKTt sc-dlnjPT cUoMmY"
                        onClick={[Function]}
                        style={Object {}}
                      >
                        3
                      </div>
                    </StyledBox>
                  </Component>
                </Box>
              </StyledPaginationButton>
            </PaginationButton>
            <PaginationButton
              active={false}
              className=""
              key="3/.3"
              onClick={[Function]}
              style={Object {}}
            >
              <StyledPaginationButton
                active={false}
                className=""
                onClick={[Function]}
                style={Object {}}
              >
                <Box
                  active={false}
                  className="sc-dlnjPT gKiNcx"
                  onClick={[Function]}
                  style={Object {}}
                >
                  <Component
                    active={false}
                    className="sc-dlnjPT gKiNcx"
                    onClick={[Function]}
                    palette={
                      Object {
                        "actionsDropdown": Object {
                          "backgroundColor": "#3fbaca",
                          "fontColor": "#ffffff",
                        },
                        "anchor": Object {
                          "disabledFontColor": "#9d9d9d",
                          "fontColor": "#3fbaca",
                        },
                        "button": Object {
                          "backgroundColor": "#3fbaca",
                          "disabledBackgroundColor": "#e2e2e2",
                          "fontColor": "#ffffff",
                        },
                        "commonElement": Object {
                          "backgroundColor": "#ffffff",
                          "borderColor": "#e2e2e2",
                          "fontColor": "#293346",
                        },
                        "description": Object {
                          "fontColor": "#9d9d9d",
                        },
                        "label": Object {
                          "fontColor": "#2d2d2d",
                        },
                        "layout": Object {
                          "backgroundColor": "#ffffff",
                          "borderColor": "#e2e2e2",
                          "fontColor": "#293346",
                        },
                        "main": Object {
                          "backgroundColor": "#f7f7f7",
                        },
                        "toggleElement": Object {
                          "backgroundColor": "#425579",
                          "borderColor": "#e2e2e2",
                          "fontColor": "#ffffff",
                          "secondaryBackgroundColor": "#ffffff",
                          "secondaryFontColor": "#9d9d9d",
                        },
                      }
                    }
                    style={Object {}}
                  >
                    <StyledBox
                      active={false}
                      className="sc-dlnjPT gKiNcx"
                      onClick={[Function]}
                      palette={
                        Object {
                          "actionsDropdown": Object {
                            "backgroundColor": "#3fbaca",
                            "fontColor": "#ffffff",
                          },
                          "anchor": Object {
                            "disabledFontColor": "#9d9d9d",
                            "fontColor": "#3fbaca",
                          },
                          "button": Object {
                            "backgroundColor": "#3fbaca",
                            "disabledBackgroundColor": "#e2e2e2",
                            "fontColor": "#ffffff",
                          },
                          "commonElement": Object {
                            "backgroundColor": "#ffffff",
                            "borderColor": "#e2e2e2",
                            "fontColor": "#293346",
                          },
                          "description": Object {
                            "fontColor": "#9d9d9d",
                          },
                          "label": Object {
                            "fontColor": "#2d2d2d",
                          },
                          "layout": Object {
                            "backgroundColor": "#ffffff",
                            "borderColor": "#e2e2e2",
                            "fontColor": "#293346",
                          },
                          "main": Object {
                            "backgroundColor": "#f7f7f7",
                          },
                          "toggleElement": Object {
                            "backgroundColor": "#425579",
                            "borderColor": "#e2e2e2",
                            "fontColor": "#ffffff",
                            "secondaryBackgroundColor": "#ffffff",
                            "secondaryFontColor": "#9d9d9d",
                          },
                        }
                      }
                      style={Object {}}
                    >
                      <div
                        className="sc-bdnylx dOPKTt sc-dlnjPT gKiNcx"
                        onClick={[Function]}
                        style={Object {}}
                      >
                        4
                      </div>
                    </StyledBox>
                  </Component>
                </Box>
              </StyledPaginationButton>
            </PaginationButton>
            <PaginationButton
              active={false}
              className=""
              key="4/.4"
              onClick={[Function]}
              style={Object {}}
            >
              <StyledPaginationButton
                active={false}
                className=""
                onClick={[Function]}
                style={Object {}}
              >
                <Box
                  active={false}
                  className="sc-dlnjPT gKiNcx"
                  onClick={[Function]}
                  style={Object {}}
                >
                  <Component
                    active={false}
                    className="sc-dlnjPT gKiNcx"
                    onClick={[Function]}
                    palette={
                      Object {
                        "actionsDropdown": Object {
                          "backgroundColor": "#3fbaca",
                          "fontColor": "#ffffff",
                        },
                        "anchor": Object {
                          "disabledFontColor": "#9d9d9d",
                          "fontColor": "#3fbaca",
                        },
                        "button": Object {
                          "backgroundColor": "#3fbaca",
                          "disabledBackgroundColor": "#e2e2e2",
                          "fontColor": "#ffffff",
                        },
                        "commonElement": Object {
                          "backgroundColor": "#ffffff",
                          "borderColor": "#e2e2e2",
                          "fontColor": "#293346",
                        },
                        "description": Object {
                          "fontColor": "#9d9d9d",
                        },
                        "label": Object {
                          "fontColor": "#2d2d2d",
                        },
                        "layout": Object {
                          "backgroundColor": "#ffffff",
                          "borderColor": "#e2e2e2",
                          "fontColor": "#293346",
                        },
                        "main": Object {
                          "backgroundColor": "#f7f7f7",
                        },
                        "toggleElement": Object {
                          "backgroundColor": "#425579",
                          "borderColor": "#e2e2e2",
                          "fontColor": "#ffffff",
                          "secondaryBackgroundColor": "#ffffff",
                          "secondaryFontColor": "#9d9d9d",
                        },
                      }
                    }
                    style={Object {}}
                  >
                    <StyledBox
                      active={false}
                      className="sc-dlnjPT gKiNcx"
                      onClick={[Function]}
                      palette={
                        Object {
                          "actionsDropdown": Object {
                            "backgroundColor": "#3fbaca",
                            "fontColor": "#ffffff",
                          },
                          "anchor": Object {
                            "disabledFontColor": "#9d9d9d",
                            "fontColor": "#3fbaca",
                          },
                          "button": Object {
                            "backgroundColor": "#3fbaca",
                            "disabledBackgroundColor": "#e2e2e2",
                            "fontColor": "#ffffff",
                          },
                          "commonElement": Object {
                            "backgroundColor": "#ffffff",
                            "borderColor": "#e2e2e2",
                            "fontColor": "#293346",
                          },
                          "description": Object {
                            "fontColor": "#9d9d9d",
                          },
                          "label": Object {
                            "fontColor": "#2d2d2d",
                          },
                          "layout": Object {
                            "backgroundColor": "#ffffff",
                            "borderColor": "#e2e2e2",
                            "fontColor": "#293346",
                          },
                          "main": Object {
                            "backgroundColor": "#f7f7f7",
                          },
                          "toggleElement": Object {
                            "backgroundColor": "#425579",
                            "borderColor": "#e2e2e2",
                            "fontColor": "#ffffff",
                            "secondaryBackgroundColor": "#ffffff",
                            "secondaryFontColor": "#9d9d9d",
                          },
                        }
                      }
                      style={Object {}}
                    >
                      <div
                        className="sc-bdnylx dOPKTt sc-dlnjPT gKiNcx"
                        onClick={[Function]}
                        style={Object {}}
                      >
                        5
                      </div>
                    </StyledBox>
                  </Component>
                </Box>
              </StyledPaginationButton>
            </PaginationButton>
            <StyledPaginationShowAll
              onClick={[Function]}
            >
              <Box
                className="sc-hKFyIo dDdERR"
                onClick={[Function]}
                style={Object {}}
              >
                <Component
                  className="sc-hKFyIo dDdERR"
                  onClick={[Function]}
                  palette={
                    Object {
                      "actionsDropdown": Object {
                        "backgroundColor": "#3fbaca",
                        "fontColor": "#ffffff",
                      },
                      "anchor": Object {
                        "disabledFontColor": "#9d9d9d",
                        "fontColor": "#3fbaca",
                      },
                      "button": Object {
                        "backgroundColor": "#3fbaca",
                        "disabledBackgroundColor": "#e2e2e2",
                        "fontColor": "#ffffff",
                      },
                      "commonElement": Object {
                        "backgroundColor": "#ffffff",
                        "borderColor": "#e2e2e2",
                        "fontColor": "#293346",
                      },
                      "description": Object {
                        "fontColor": "#9d9d9d",
                      },
                      "label": Object {
                        "fontColor": "#2d2d2d",
                      },
                      "layout": Object {
                        "backgroundColor": "#ffffff",
                        "borderColor": "#e2e2e2",
                        "fontColor": "#293346",
                      },
                      "main": Object {
                        "backgroundColor": "#f7f7f7",
                      },
                      "toggleElement": Object {
                        "backgroundColor": "#425579",
                        "borderColor": "#e2e2e2",
                        "fontColor": "#ffffff",
                        "secondaryBackgroundColor": "#ffffff",
                        "secondaryFontColor": "#9d9d9d",
                      },
                    }
                  }
                  style={Object {}}
                >
                  <StyledBox
                    className="sc-hKFyIo dDdERR"
                    onClick={[Function]}
                    palette={
                      Object {
                        "actionsDropdown": Object {
                          "backgroundColor": "#3fbaca",
                          "fontColor": "#ffffff",
                        },
                        "anchor": Object {
                          "disabledFontColor": "#9d9d9d",
                          "fontColor": "#3fbaca",
                        },
                        "button": Object {
                          "backgroundColor": "#3fbaca",
                          "disabledBackgroundColor": "#e2e2e2",
                          "fontColor": "#ffffff",
                        },
                        "commonElement": Object {
                          "backgroundColor": "#ffffff",
                          "borderColor": "#e2e2e2",
                          "fontColor": "#293346",
                        },
                        "description": Object {
                          "fontColor": "#9d9d9d",
                        },
                        "label": Object {
                          "fontColor": "#2d2d2d",
                        },
                        "layout": Object {
                          "backgroundColor": "#ffffff",
                          "borderColor": "#e2e2e2",
                          "fontColor": "#293346",
                        },
                        "main": Object {
                          "backgroundColor": "#f7f7f7",
                        },
                        "toggleElement": Object {
                          "backgroundColor": "#425579",
                          "borderColor": "#e2e2e2",
                          "fontColor": "#ffffff",
                          "secondaryBackgroundColor": "#ffffff",
                          "secondaryFontColor": "#9d9d9d",
                        },
                      }
                    }
                    style={Object {}}
                  >
                    <div
                      className="sc-bdnylx dOPKTt sc-hKFyIo dDdERR"
                      onClick={[Function]}
                      style={Object {}}
                    >
                      Show all
                    </div>
                  </StyledBox>
                </Component>
              </Box>
            </StyledPaginationShowAll>
          </div>
        </StyledBox>
      </Component>
    </Box>
  </StyledPagination>
</Pagination>
`;
